src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Engine/Trivia/CSharpTriviaFormatter.cs

```diff
@@ -78,7 +78,10 @@ protected override LineColumnRule GetLineColumnRuleBetween(SyntaxTrivia trivia1,
 
                 if (insertNewLine)
                 {
-                    return LineColumnRule.PreserveLinesWithDefaultIndentation(lines: 0);
+                    if (existingWhitespaceBetween.Spaces != this.Spaces)
+                        return LineColumnRule.PreserveWithGivenSpaces(spaces: this.Spaces);
+                    else
+                        return LineColumnRule.PreserveLinesWithDefaultIndentation(lines: 0);
```

Comment:
i would doc this line.  i think i get it, but it would be nice to ahve confirmation of intent.

Reply:
➡️ This block now eliminated